_posts/2020-09-16-chross-browser-mobile-webapp.markdown

```diff
@@ -0,0 +1,127 @@
+---
+layout: post
+title: "Cross-Browser Mobile Web Apps (Snap Minis Part 2)"
+date: 2020-11-05 12:00:00
+categories: ["learning", "crossbrowser"]
+author: arieljuod
+---
+
+In keeping with [the first part of this series](https://www.ombulabs.com/blog/learning/creating-a-snapchat-mini.html), I'll list a few interesting issues we found when making the apps work as similar as possible for Android's webview and iOS' webview.
+
+<!--more-->
+
+Some of these issues are related to mobile web development in general, it's not something specific to Snap Minis but Snapchat uses the native webview components to display your Mini. Since you'll want your Mini to work properly for all users you'll have to make fixes for specific issues you may encounter for each webview implementation.
+
+Android uses Chrome by default for the webview, but it is different than desktop Chrome. iOS uses Safari for the webview, but it is also different than desktop Safari. Keep in mind you can fix some of the issues using desktop browsers but you'll have to test them in the mobile browsers to be sure.
+
+## Keypress/release/up/down Events are not Triggered for Mobile Chrome
+
+When you type using Android's default keyboard app, it won't trigger the usual key-related events (`keypress`, `keyup`, etc). Depending on your needs, you may be able to workaround this limitation using an event listener for the `input` or the `change` events.
+
+## MAXLENGTH Attribute won't Work on Mobile Chrome
+
+If you have an input where you want to limit the value's length, the simplest way to do that is to use the input's `maxlength` HTML attribute. That will limit the length disallowing the user to type more characters than expected. But, when typing on an Android device, it won't respect that limit.
+
+The first idea would be to use the `keypress` event and prevent its default behavior depending on the pressed key and the element's value length, but, since we can't relay on the `keypress` event, we had to implement another workaround using the `input` event:
```

Comment:
```suggestion
The first idea would be to use the `keypress` event and prevent its default behavior depending on the pressed key and the element's value length, but, since we can't rely on the `keypress` event, we had to implement another workaround using the `input` event:
```